File: shopapp-angular/src/app/user/models/order.detail.ts
import { Order } from "./order";
import { Product } from "./product";

export interface OrderDetail {
  id: number;
  order: Order;
  product: Product;
  price: number;
  number_of_products: number;
  total_money: number;
  color?: string; // Dấu "?" cho biết thuộc tính này là tùy chọn
}

File: ShopAppSpring/src/main/java/com/duchung/shopappspring/models/OrderDetail.java
package com.duchung.shopappspring.models;

import com.duchung.shopappspring.domains.BaseEntity;
import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.*;

@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "order_details")
@Entity
@Builder
public class OrderDetail extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "order_id")
    @JsonBackReference
    private Order order;

    @ManyToOne
    @JoinColumn(name = "product_id")
    private Product product;

    private double price;

    @JsonProperty("number_of_products")
    private int numberOfProducts;

    @JsonProperty("total_money")
    private double totalMoney;

    @Column(length = 20)
    private String color;

    private int status;



}
